TypeCobol.LanguageServer/WorkspaceProjectStore.cs

```diff
@@ -0,0 +1,286 @@
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Update for each WorkspaceProject instance its compilation options and AnalyzerProvider instance
+        /// </summary>
+        /// <param name="compilationOptions">New Compilation Options</param>
+        /// <param name="analyzerProvider">new Analyzer Provider instance</param>
+        internal void UpdateProjectOptionsAndAnalyzerProvider(TypeCobolOptions compilationOptions, IAnalyzerProvider analyzerProvider)
+        {
+            foreach (var wksProject in this)
+            {
+                wksProject.UpdateProjectOptionsAndAnalyzerProvider(compilationOptions, analyzerProvider);
+            }
+        }
+
+        /// <summary>
+        /// Refresh all documents in all WorkspaceProject instances
+        /// </summary>
+        /// <param name="workspace">The main Workspace instance</param>
+        internal void DoRefreshOpenedFiles(Workspace workspace)
+        {
+            foreach (var wksProject in this)
+            {
+                wksProject.DoRefreshOpenedFiles(workspace);
+            }
+        }
+
+        /// <summary>
+        /// Move a Document from one WorkspaceProject to another WorkSpaceProject.
+        /// </summary>
+        /// <param name="docContext">The Document Context instance to move</param>
+        /// <param name="fromWorkspaceProject">The Source WorkspaceProject instance</param>
+        /// <param name="toWorkspaceProject">The Target WorkspaceProject instance.</param>
+        internal void MoveWorkspaceProjectDocument(DocumentContext docContext, WorkspaceProject fromWorkspaceProject, WorkspaceProject toWorkspaceProject)
+        {
+            System.Diagnostics.Debug.Assert(docContext != null && fromWorkspaceProject != null && toWorkspaceProject != null);
+            System.Diagnostics.Debug.Assert(fromWorkspaceProject.Contains(docContext));
+            System.Diagnostics.Debug.Assert(!toWorkspaceProject.Contains(docContext));
+
+            fromWorkspaceProject.RemoveDocument(docContext);
+            toWorkspaceProject.AddDocument(docContext);            
+        }
+
+        /// <summary>
+        /// Associate a document to a Workspace Project
+        /// </summary>
+        /// <param name="docUri">The Uri of the document to be associated to workspace projet</param>
+        /// <param name="projectKey">The project key to which the document shall be associated</param>
+        /// <param name="copyFolders">List of Copy Folders to be associated with the underlying CompilationProject instance.</param>
+        /// <param name="workspace">The Workspace project instance</param>
+        internal void AssociateDocumentToProject(Uri docUri, string projectKey, List<string> copyFolders, Workspace workspace)
+        {
+            // Check if the docUri is already associated to a WorkspaceProject
+            DocumentContext docContext = null;
+            WorkspaceProject workspaceProject = null;
```

Comment:
Can be inlined

Reply:
Changed in [e1c296c](https://github.com/TypeCobolTeam/TypeCobol/pull/2145/commits/e1c296c5835196790db878e8efdab69f6a196c18)